packages/amplify_storage_s3/android/src/main/kotlin/com/amazonaws/amplify/amplify_storage_s3/AmplifyStorageOperations.kt

```diff
@@ -197,15 +186,30 @@ class AmplifyStorageOperations {
             }
         }
 
-        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull msg: String, localizedError: String?, recoverySuggestion: String?) {
-            var errorDetails = HashMap<String, Any>();
-            errorDetails.put("PLATFORM_EXCEPTIONS", mapOf(
-                    "platform" to "Android",
-                    "localizedErrorMessage" to localizedError,
-                    "recoverySuggestion" to recoverySuggestion
-            ))
+        private fun prepareError(@NonNull flutterResult: MethodChannel.Result, @NonNull e: Exception) {
+            val errorCode = "StorageException"
+            var message = ExceptionMessages.unexpectedExceptionMessage
+            var recoverySuggestion = ExceptionMessages.unexpectedExceptionSuggestion
+            LOG.error(errorCode, e)
+            val serializedError: Map<String, Any?> = if (e is StorageException) {
+                ExceptionUtil.createSerializedError(e)
+            } else if (e is AmplifyFlutterValidationException) {
+                message = e.message ?: message
+                recoverySuggestion = e.recoverySuggestion
```

Comment:
I do recommend sending unknown/unexpected messages for validation errors found in native (with `underlyingException` containing the full validation error/message).

Reply:
Fixed.